src/PortingAssistantExtensionTelemetry/Utils/LogUploadUtils.cs

```diff
@@ -49,48 +49,63 @@ private static bool IsFileLocked(FileInfo file)
             string logName,
             string logData,
             string profile,
+            bool enabledDefaultCredentials,
             TelemetryConfiguration telemetryConfiguration
             )
         {
             try
             {
                 var chain = new CredentialProfileStoreChain();
                 AWSCredentials awsCredentials;
-                var profileName = profile;
-                var region = telemetryConfiguration.Region;
-                if (chain.TryGetAWSCredentials(profileName, out awsCredentials))
+
+                if (enabledDefaultCredentials)
+                {
+                    awsCredentials = FallbackCredentialsFactory.GetCredentials();
+                    if (awsCredentials == null)
+                    {
+                        Console.WriteLine("Invalid Credentials.");
+                        return false;
+                    }
+                }
+                else
                 {
-                    dynamic requestMetadata = new JObject();
-                    requestMetadata.version = "1.0";
-                    requestMetadata.service = telemetryConfiguration.ServiceName;
-                    requestMetadata.token = "12345678";
-                    requestMetadata.description = telemetryConfiguration.Description;
+                    var profileName = profile;
+                    if (!chain.TryGetAWSCredentials(profileName, out awsCredentials))
+                    {
+                        Console.WriteLine("Invalid Credentials.");
+                        return false;
+                    }
+                }
```

Comment:
+1 on my suggestion for a "GetCreds" method that maybe could be used here as well?